src/lib/firebaseBackend.js

```diff
@@ -18,44 +19,317 @@ class FirebaseBackend {
           dict["id"] = d.id;
           return dict;
         });
-        console.log(data);
+        return data;
       })
       .catch(console.log);
     // To do
+    // create zipcode and radius filters
   }
 
   getDropSites(dropSiteId) {
-    // To do
+    if (dropSiteId) {
+      return this.firestore
+        .collection("dropSite")
+        .doc(dropSiteId)
+        .get()
+        .then(doc => {
+          return doc.data();
+        })
+        .catch(console.log);
+    } else {
+      console.log("Error, one or more required params missing.");
+      return Promise.resolve("Error, one or more required params missing.");
+    }
   }
 
-  addDropSite(dropSiteName, location_id, dropAddress, arbitraryComments) {
-    // To do
+  addDropSite(
+    location_id,
+    dropSiteName,
+    dropSiteDescription,
+    dropSiteAddress,
+    dropSiteZip
+  ) {
+    if (dropSiteName && location_id && dropSiteAddress && dropSiteZip) {
+      let newSiteObj = {
+        dropSiteName: dropSiteName,
+        location_id: location_id,
+        dropSiteAddress: dropSiteAddress,
+        dropSiteZip: dropSiteZip
+      };
+      if (dropSiteDescription) {
+        newSiteObj.dropSiteDescription = dropSiteDescription;
+      }
+      return this.firestore
+        .collection("dropSite")
+        .doc(location_id)
+        .set(newSiteObj)
+        .then(function(docRef) {
+          return "Drop site added";
+        })
+        .catch(function(error) {
+          console.error("Error writing document: ", error);
+        });
+    } else {
+      console.log("Error, one or more required params missing.");
+      return Promise.resolve("Error, one or more required params missing.");
+    }
+  }
+
+  editDropSite(
+    location_id,
+    dropSiteName,
+    dropSiteDescription,
+    dropSiteAddress,
+    dropSiteZip
+  ) {
+    if (
+      location_id &&
+      (dropSiteName || dropSiteDescription || dropSiteAddress || dropSiteZip)
+    ) {
+      let newSiteObj = {};
+      if (dropSiteName) {
+        newSiteObj.dropSiteName = dropSiteName;
+      }
+      if (dropSiteDescription) {
+        newSiteObj.dropSiteDescription = dropSiteDescription;
+      }
+      if (dropSiteAddress) {
+        newSiteObj.dropSiteAddress = dropSiteAddress;
+      }
+      if (dropSiteZip) {
+        newSiteObj.dropSiteZip = dropSiteZip;
+      }
+      return this.firestore
+        .collection("dropSite")
+        .doc(location_id)
+        .set(newSiteObj, { merge: true })
+        .then(function() {
+          return "Drop site updated";
+        })
+        .catch(function(error) {
+          console.error("Error writing document: ", error);
+        });
+    } else {
+      console.log("Error, one or more required params missing.");
+      return Promise.resolve("Error, one or more required params missing.");
```

Comment:
I think it's more idiomatic to do Promise.reject(new Error('message')). (Although it looks like we aren't catching any errors at the moment, so it wouldn't actively change any behavior)